src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -522,6 +530,10 @@ export abstract class VerifiableCredentialsServiceBase {
     return pex.selectFrom(presentationDefinition, vc);
   }
 
+  public filterSelfSignDescriptors(descriptors) {
```

Comment:
typing for input_descriptors is not available for import from sphereon (InputDescriptorV1 | InputDescriptorV2)

Reply:
IMO, it is very important to provide a comment about why we are doing this as it could be very hard from someone reading the code in the future to know why this was done.
```suggestion
  /*
   * We have observed that pex may have bugs when dealing with subject_is_issuer credentials
   * and when handling a presentation definition with more than one input descriptor.
   * This could be related to these issues:
   *    - https://github.com/Sphereon-Opensource/pex/issues/96
   *    - https://github.com/Sphereon-Opensource/pex/issues/91
   * We are therefore trying to simplify the input to pex so remove the possibility of it generating incorrect results.
   * Once the above issues are fixed, pex can be updated and perhaps this filtering will not needed.
   */
  public filterSelfSignDescriptors(descriptors) {
```